crates/ads126x/src/register.rs

```diff
@@ -0,0 +1,123 @@
+use bitflags::bitflags;
+
+#[repr(u8)]
+pub enum Register {
+    ID = 0x00,
+    POWER = 0x01,
+    INTERFACE = 0x02,
+    MODE0 = 0x03,
+    MODE1 = 0x04,
+    MODE2 = 0x05,
+    INPMUX = 0x06,
+    OFCAL0 = 0x07,
+    OFCAL1 = 0x08,
+    OFCAL2 = 0x09,
+    FSCAL0 = 0x0A,
+    FSCAL1 = 0x0B,
+    FSCAL2 = 0x0C,
+    IDACMUX = 0x0D,
+    IDACMAG = 0x0E,
+    REFMUX = 0x0F,
+    TDACP = 0x10,
+    TDACN = 0x11,
+    GPIOCON = 0x12,
+    GPIODIR = 0x13,
+    GPIODAT = 0x14,
+    ADC2CFG = 0x15,
+    ADC2MUX = 0x16,
+    ADC2OFC0 = 0x17,
+    ADC2OFC1 = 0x18,
+    ADC2FSC0 = 0x19,
+    ADC2FSC1 = 0x1A,
+}
+
+bitflags! {
+    pub struct StatusRegister: u8 {
+        const ADC2      = 0b1000_0000;
+        const ADC1      = 0b0100_0000;
+        const EXTCLK    = 0b0010_0000;
+        const REF_ALM   = 0b0001_0000;
+        const PGAL_ALM  = 0b0000_1000;
+        const PGAH_ALM  = 0b0000_0100;
+        const PGAD_ALM  = 0b0000_0010;
+        const RESET     = 0b0000_0001;
+    }
+}
+
+impl StatusRegister {
```

Comment:
Are these implementations needed since it is just a `.contains(StatusRegister::*)` call? If we want to go with `.is_*()` then we should implement this for all the registers. It seems excessive, but it will just get inlined anyway, so no performance cost.